browser/ui/views/translate/brave_translate_bubble_view.cc

```diff
@@ -0,0 +1,143 @@
+                0);
+  cs->AddPaddingColumn(
+      views::GridLayout::kFixedSize,
+      provider->GetDistanceMetric(views::DISTANCE_RELATED_BUTTON_HORIZONTAL));
+  cs->AddColumn(views::GridLayout::LEADING, views::GridLayout::CENTER,
+                views::GridLayout::kFixedSize, views::GridLayout::USE_PREF, 0,
+                0);
+
+  auto accept_button = views::MdTextButton::CreateSecondaryUiButton(
+      this, l10n_util::GetStringUTF16(IDS_BRAVE_TRANSLATE_BUBBLE_INSTALL));
+  accept_button->SetID(BUTTON_ID_TRANSLATE);
+  accept_button->SetIsDefault(true);
+
+  auto cancel_button = views::MdTextButton::CreateSecondaryUiButton(
+      this, l10n_util::GetStringUTF16(IDS_BRAVE_TRANSLATE_BUBBLE_CANCEL));
+  cancel_button->SetID(BUTTON_ID_CANCEL);
+
+  layout->StartRowWithPadding(
+      views::GridLayout::kFixedSize, kButtonColumnSetId,
+      views::GridLayout::kFixedSize,
+      provider->GetDistanceMetric(views::DISTANCE_UNRELATED_CONTROL_VERTICAL));
+
+  if (views::PlatformStyle::kIsOkButtonLeading) {
+    layout->AddView(accept_button.release());
+    layout->AddView(cancel_button.release());
+  } else {
+    layout->AddView(cancel_button.release());
+    layout->AddView(accept_button.release());
+  }
+
+  return view;
+}
+
+void BraveTranslateBubbleView::InstallGoogleTranslate() {
+  Browser* browser = chrome::FindBrowserWithWebContents(web_contents());
+  if (!browser)
+    return;
+
+  BraveTranslateIconView* translate_icon =
+    static_cast<BraveTranslateIconView*>(
+        BrowserView::GetBrowserViewForBrowser(browser)
+        ->toolbar_button_provider()
+        ->GetOmniboxPageActionIconContainerView()
+        ->GetPageActionIconView(PageActionIconType::kTranslate));
+  DCHECK(translate_icon);
+
+  translate_icon->InstallGoogleTranslate();
+}
+
+void BraveTranslateBubbleView::ButtonPressed(views::Button* sender,
+                                             const ui::Event& event) {
+  switch (static_cast<ButtonID>(sender->GetID())) {
+    case BUTTON_ID_TRANSLATE: {
+      InstallGoogleTranslate();
+      break;
+    }
+    case BUTTON_ID_CANCEL: {
+      CloseBubble();
+      break;
+    }
+    default: {
+      // We don't expect other buttons used by chromium's original views.
+      NOTREACHED();
+    }
+  }
+}
+
+bool BraveTranslateBubbleView::AcceleratorPressed(
+    const ui::Accelerator& accelerator) {
+  switch (model_->GetViewState()) {
+    case TranslateBubbleModel::VIEW_STATE_BEFORE_TRANSLATE: {
+      if (accelerator.key_code() == ui::VKEY_RETURN) {
+        InstallGoogleTranslate();
+        return true;
+      }
+      break;
+    }
+    default: {
+      // We don't expect views in other states.
+      NOTREACHED();
+    }
+  }
+
+  return TranslateBubbleView::AcceleratorPressed(accelerator);
+}
+
+void BraveTranslateBubbleView::Init() {
+  TranslateBubbleView::Init();
+  RemoveChildView(before_translate_view_);
+  before_translate_view_ = AddChildView(BraveCreateViewBeforeTranslate());
```

Comment:
I think this is going to fail on android because it has a guard above

Reply:
it seems like in general this needs to work differently for android

Reply:
this google extension approach won't work in android unless we have extensions supported, so this code block shouldn't be built on android atm.